src/standalone/Vegetation/Canopy.jl

```diff
@@ -693,9 +694,18 @@ function CanopyModel{FT}(;
     end
 
     if typeof(hydraulics) <: PlantHydraulicsModel{FT}
-        @assert biomass.height ==
-                hydraulics.compartment_surfaces[end] -
-                hydraulics.compartment_surfaces[1]
+        hydraulics_height =
+            hydraulics.compartment_surfaces[end] -
+            hydraulics.compartment_surfaces[1]
+        # For scalar height: direct comparison
+        # For Field height: check all values match (with tolerance for floating point)
+        if biomass.height isa ClimaCore.Fields.Field
+            @assert all(
+                isapprox.(biomass.height, hydraulics_height, rtol = 1e-10),
+            ) "Biomass height Field must match hydraulics height"
+        else
+            @assert biomass.height ≈ hydraulics_height "Biomass height must match hydraulics height"
```

Comment:
I dont quite understand how this works, because the hydraulics_height is a scalar. How can this pass when biomass.height is spatialy varying?

In general we need to go through src and make sure these are consistent, because in some places we use the 'hydraulics" height as the height

Reply:
You're right - the hydraulics compartments are always scalar (1m). When biomass.height varies spatially, only the aerodynamic calculations use those varying heights. The hydraulics structure stays uniform everywhere.

This is a current limitation - we can extend hydraulics to support spatial variation in a follow-up PR.